tests/utilities/test_model_helpers.py

```diff
@@ -37,13 +37,40 @@ class TestModel(BoringModel):
         def foo(self):
             pass
 
+        def bar(self):
+            return 1
+
     with pytest.raises(ValueError, match="The parent should define the method"):
         is_overridden("foo", TestModel())
 
     # normal usage
     assert is_overridden("training_step", model)
     assert is_overridden("train_dataloader", datamodule)
 
+    class WrappedModel(TestModel):
+
+        def __new__(cls, *args, **kwargs):
+            obj = super().__new__(cls)
+            obj.foo = cls.wrap(obj.foo)
+            obj.bar = cls.wrap(obj.bar)
+            return obj
+
+        @staticmethod
+        def wrap(fn):
+
+            @wraps(fn)
+            def wrapper():
+                fn()
+
+            return wrapper
+
+        def bar(self):
+            return 2
+
+    # `functools.wraps()` support
+    assert not is_overridden("foo", WrappedModel(), parent=TestModel)
```

Comment:
I feel like this shall mark as overwritten as a wrapped can significantly change its functionality...

Reply:
The current implementation assumes that the wrapper will be the same both for the instance and the parent

Not sure whether we could check if the wrappers are different

Reply:
can you detect what wrapper was used and then compare the wrapper itself?

Reply:
Can detect that a wrapper was used (current PR changes) but no idea about comparing the wrappers

Reply:
Could we check if the wrapper is from `Lightning` or `users` ?

Reply:
Might be worth for another PR. I believe this is a pretty rare use-case.